aerogear/mobile-security-service-operator
# fix(DB) - Add timer.sleep for controller be able to get the configMap created by the App CRD (AEROGEAR-9084)

## Motivation
https://issues.jboss.org/browse/AEROGEAR-9084

## What
* Delay the DB CRD controller create the deployment for it be able to get the configMap created by the APP CRD controller
* Fix the func to check it since it was not passing the spec name 
* Improve the comments to make clear how it is working
* Making database env values optional in the DB CR spec

## Why
* The DB and APP should use the data/values defined in the ConfigMap as default and just create the env variables with the spec values from the DB CR if the ConfigMap was not found.

## Verification Steps

1. Build, publish, install and config the operator locally with a dev image based on this change
2. Run `make create-all`
3. Check that all finished with success and that the ENV vars from the DB are from the ConfigMap
<img width="1583" alt="Screenshot 2019-04-17 at 12 27 37" src="https://user-images.githubusercontent.com/7708031/56284840-a0b20080-610d-11e9-99a5-4bed0b93c055.png">

## Checklist:


## Progress
